aiidalab_widgets_base/process.py

```diff
@@ -21,9 +30,9 @@ def get_running_calcs(process):
     # If the process is a running work chain - returning its children
     if issubclass(type(process), WorkChainNode) and not process.is_sealed:
         calcs = []
-        for link in process.get_outgoing():
-            if issubclass(type(link.node), ProcessNode) and not link.node.is_sealed:
-                calcs += get_running_calcs(link.node)
+        for out_link in process.get_outgoing():
+            if issubclass(type(out_link.node), ProcessNode) and not out_link.node.is_sealed:
+                calcs += get_running_calcs(out_link.node)
         return calcs
```

Comment:
Why not make this is a generator function?

Reply:
Changed in c540efe. The only thing I am not sure about is this:
https://github.com/aiidalab/aiidalab-widgets-base/blob/c540efee4cb0e50cedec1593e3b0567f5ee7e39d/aiidalab_widgets_base/process.py#L36

Do you know of any better way to return a "falsy" iterator?


I also took this occasion to redesign `RunningCalcJobOutputWidget` to allow selecting a running sub-process.

Reply:
It is not possible to return anything from a generator function. Could you briefly explain under what circumstances it would be important to check the return value? I can then probably recommend a good solution.

Reply:
I would like to ensure, that `get_running_calcs` does 0 iterations in case it receives not a process object. In other words:
```
res = list(get_running_calcs(not_a_process)) 
```
`res` should become an empty list.

Reply:
ah, great, so in my case I should just remove the `return []`

Reply:
done